src/bootstrap/src/core/build_steps/gcc.rs

```diff
@@ -0,0 +1,222 @@
+//             // the GCC shared object file is named `gcc-12-rust-{version}-nightly`
+//             config.src.join("src/version"),
+//         ]);
+//         output(&mut rev_list).trim().to_owned()
+//     } else if let Some(info) = channel::read_commit_info_file(&config.src) {
+//         info.sha.trim().to_owned()
+//     } else {
+//         "".to_owned()
+//     };
+
+//     if gcc_sha.is_empty() {
+//         eprintln!("error: could not find commit hash for downloading GCC");
+//         eprintln!("HELP: maybe your repository history is too shallow?");
+//         eprintln!("HELP: consider disabling `download-ci-gcc`");
+//         eprintln!("HELP: or fetch enough history to include one upstream commit");
+//         panic!();
+//     }
+
+//     gcc_sha
+// }
+
+// /// Returns whether the CI-found GCC is currently usable.
+// ///
+// /// This checks both the build triple platform to confirm we're usable at all,
+// /// and then verifies if the current HEAD matches the detected GCC SHA head,
+// /// in which case GCC is indicated as not available.
+// pub(crate) fn is_ci_gcc_available(config: &Config, asserts: bool) -> bool {
+//     let supported_platforms = [
+//         // tier 1
+//         ("x86_64-unknown-linux-gnu", true),
+//     ];
+
+//     if !supported_platforms.contains(&(&*config.build.triple, asserts))
+//         && (asserts || !supported_platforms.contains(&(&*config.build.triple, true)))
+//     {
+//         return false;
+//     }
+
+//     if is_ci_gcc_modified(config) {
+//         eprintln!("Detected GCC as non-available: running in CI and modified GCC in this change");
+//         return false;
+//     }
+
+//     true
+// }
+
+// /// Returns true if we're running in CI with modified GCC (and thus can't download it)
+// pub(crate) fn is_ci_gcc_modified(config: &Config) -> bool {
+//     CiEnv::is_ci() && config.rust_info.is_managed_git_subrepository() && {
+//         // We assume we have access to git, so it's okay to unconditionally pass
+//         // `true` here.
+//         let gcc_sha = detect_gcc_sha(config, true);
+//         let head_sha = output(config.git().arg("rev-parse").arg("HEAD"));
+//         let head_sha = head_sha.trim();
+//         gcc_sha == head_sha
+//     }
+// }
+
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+pub struct Gcc {
+    pub target: TargetSelection,
+}
+
+impl Step for Gcc {
+    type Output = bool;
+
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.path("src/gcc")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(Gcc { target: run.target });
+    }
+
+    /// Compile GCC for `target`.
+    fn run(self, builder: &Builder<'_>) -> bool {
+        let target = self.target;
+        if !target.contains("linux") || !target.contains("x86_64") {
+            return false;
+        }
+
+        // If GCC has already been built or been downloaded through download-ci-gcc, we avoid
```

Comment:
There is no `download-ci-gcc` yet, this comment is thus outdated (or maybe future-dated :) ).

Reply:
Future-dated for sure. I was having sight maybe too far from present time. :sweat_smile: